diff --git a/clutter/wayland/clutter-backend-wayland.c b/clutter/wayland/clutter-backend-wayland.c.orig
index 77ed8c4..542c2b8 100644
--- a/clutter/wayland/clutter-backend-wayland.c
+++ b/clutter/wayland/clutter-backend-wayland.c.orig
@@ -238,10 +238,6 @@ clutter_backend_wayland_get_renderer (ClutterBackend  *backend,
 
   cogl_wayland_renderer_set_foreign_display (renderer,
                                              backend_wayland->wayland_display);
-  cogl_wayland_renderer_set_foreign_compositor (renderer,
-                                                backend_wayland->wayland_compositor);
-  cogl_wayland_renderer_set_foreign_shell (renderer,
-                                           backend_wayland->wayland_shell);
 
   return renderer;
 }
